docs/reference-guides/best-practices/rancher-server/tuning-and-best-practices-for-rancher-at-scale.md

```diff
@@ -0,0 +1,94 @@
+
+* Reduce network latency between the upstream Rancher cluster and downstream clusters to the extent possible. Note that latency is, among other factors, a function of geographic distance - if a you require clusters or nodes spread across the world, consider multiple Rancher installations.
+
+## Minimizing Load on the Upstream Cluster
+
+When scaling up Rancher, one typical bottleneck is resource growth in the upstream (local) Kubernetes cluster. The upstream cluster contains information for all downstream clusters. Many operations that apply to downstream clusters create new objects in the upstream cluster and require computation from handlers running in the upstream cluster.
+
+### Managing Your Object Counts
+
+Etcd is the backing database for Kubernetes and for Rancher. The database may eventually encounter limitations to the number of a single Kubernetes resource type it can store. Exact limits vary and depend on a number of factors. However, experience indicates that performance issues frequently arise once a single resource type's object count exceeds 60 thousand. Often that type is `RoleBinding`.
+
+This is typical in Rancher, as many operations create new `RoleBinding` objects in the upstream cluster as a side effect.
+
+You can reduce the number of `RoleBindings` in the upstream cluster in the following ways:
+* Limit the use of the [Restricted Admin](../../../how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/manage-role-based-access-control-rbac/global-permissions#restricted-admin) role. Apply other roles wherever possible.
+* If you use [external authentication](../../../pages-for-subheaders/authentication-config), use groups to assign roles.
+* Only add users to clusters and projects when necessary.
+* Remove clusters and projects when they are no longer needed.
+* Only use custom roles if necessary.
+* Use as few rules as possible in custom roles.
+* Consider whether adding a role to a user is redundant.
+* Consider using less, but more powerful, clusters.
+* Kubernetes permissions are always "additive" (allow-list) rather than "subtractive" (deny-list). Try to minimize configurations that gives access to all but one aspect of a cluster, project, or namespace, as that will result in the creation of a high number of `RoleBinding` objects.
+* Experiment to see if creating new projects or clusters manifests in fewer `RoleBindings` for your specific use case.
+
+### RoleBinding Count Estimation
+
+Predicting how many `RoleBinding` objects a given configuration will create is complicated. However, the following considerations can offer a rough estimate:
+* For a minimum estimate, use the formula `32C + U + 2UaC + 8P + 5Pa`, where `C` is the total number of clusters, `U` is the total number of users, `Ua` is the average number of users with a membership on a cluster, `P` is the total number of projects, and `Pa` is the average number of users with a membership on a project.
```

Comment:
```suggestion
* For a minimum estimate, use the formula `32C + U + 2UaC + 8P + 5Pa`. 
  * `C` is the total number of clusters.
  * `U` is the total number of users.
  * `Ua` is the average number of users with a membership on a cluster.
  * `P` is the total number of projects.
  * `Pa` is the average number of users with a membership on a project.
```

Spreading the definitions over multiple lines for readability, similar what line 64-66 has already done,